src/data/lock.ts

```diff
@@ -37,7 +37,10 @@ export const callProtectedLockService = async (
   ).catch(() => undefined);
   const defaultCode = lockRegistryEntry?.options?.lock?.default_code;
 
-  if (stateObj!.attributes.code_format && !defaultCode) {
+  if (
+    stateObj!.attributes.code_format &&
+    (defaultCode == null || defaultCode === "")
```

Comment:
This will still show the dialog when `defaultCode` is `""`? What is exactly the goal?

Reply:
If there is a default code in the entity registry, user should not be prompted to enter a code because the back-end will automatically send the code for unlocking.
Just checking `!defaultCode` should be enough right?

Reply:
I thought it could also be an empty string? Or did I misunderstood?

Reply:
nvm

Reply:
In core it sets the default_code to `""` when the entity is registered in the entity registry but it can also be entities without unique id so we also need to handle `null´

Reply:
yes